daft/dataframe/dataframe.py

```diff
@@ -158,6 +158,28 @@ def _result(self) -> Optional[PartitionSet]:
         else:
             return self._result_cache.value
 
+    def explain_broadcast(self):
+        """Broadcast the mermaid-formatted plan on the given port (assuming metrics-broadcasting is enabled)."""
+        import socket
+
+        from daft.dataframe.display import MermaidFormatter
+
+        ctx = get_context()
+        with ctx._lock:
+            if not ctx._enable_broadcast:
+                return
+
+            addr = ctx._broadcast_addr
+            port = ctx._broadcast_port
+
+        is_cached = self._result_cache is not None
+        instance = MermaidFormatter(builder=self.__builder, show_all=True, simple=False, is_cached=is_cached)
+        text: str = instance._repr_markdown_()
+
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as sock:
+            sock.connect((addr, port))
```

Comment:
Should we not just make this a REST endpoint that we can send a normal JSON PUT request to? Seems simpler.

Reply:
Ya, either ways will suffice, but using the HTTP application protocol over TCP (as opposed to raw TCP), will help us down the road when the broadcasting logic becomes more involved.

Updated to use HTTP broadcasting instead of raw TCP broadcasting!